File: Remove Nth Node From End of List.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* fst=head;ListNode* slw=head;
        if (n==0) return head;
        for(int i=0;i<n;i++){
            fst=fst->next;
        }
        if(fst == NULL ) return head->next;
        //cout << slw->val;
        while(fst->next != NULL){
            //cout <<"hello";
            slw=slw->next;
            fst=fst->next;
        }
        
        slw->next=(slw->next)->next ;
        
     return head;   
    }
};

File: Leetcode -14 Days Study Plan to Crack Algo (Solutions)/Study Plan 1 - (Algorithm I)/Day 5/Java/Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        ListNode slow = head;
        ListNode fast = head;
        
        int i = 0;
        
        while(i<n){
            fast = fast.next;
            i++;
        }
        
        if(fast == null) return head.next;
        
         while(fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next;
        }
        
        slow.next = slow.next.next;
        
        return head;
    }
}